MDANSE/Src/MDANSE/Framework/Configurators/IConfigurator.py

```diff
@@ -145,6 +154,16 @@ def __init__(self, name, **kwargs):
 
         self._original_input = ""
 
+    def __str__(self) -> str:
+        return "\n".join(
+            [""]
+            + [
+                f"{label}={str(getattr(self, label, 'Not set'))}"
+                for label in self._printable_attributes
+            ]
+            + [f"{key}={str(self.get(key, 'Not set'))}" for key in self.keys()]
```

Comment:
Maybe a usecase for [`value_chain`](https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.value_chain) is this more readable?
```suggestion
        return "\n".join(
            value_chain(
                "",
                (
                    f"{label}={str(getattr(self, label, 'Not set'))}"
                    for label in self._printable_attributes
                ),
                (f"{key}={str(self.get(key, 'Not set'))}" for key in self),
            )
        )
```